poll/poll_test.go

```diff
@@ -38,6 +40,52 @@ func TestWaitOn(t *testing.T) {
 	assert.Equal(t, end, counter)
 }
 
+func TestWaitOnFile(t *testing.T) {
+	fakeT := &fakeT{}
+	fakeFilePath := "./fakefile"
+	defer os.Remove(fakeFilePath)
+
+	go func() {
+		time.Sleep(time.Second)
+		os.Create(fakeFilePath)
+	}()
+
+	_, err := os.Stat(fakeFilePath)
+	assert.ErrorType(t, err, os.IsNotExist)
+	WaitOn(fakeT, FileCheck(fakeFilePath))
+	_, err = os.Stat(fakeFilePath)
+	assert.NilError(t, err)
+}
+
+func TestWaitOnSocketWithTimeout(t *testing.T) {
+	fakeT := &fakeT{}
+	network := "tcp"
+	fakeURL := "localhost:55555"
+	done := make(chan struct{})
+
+	go func(done chan struct{}) {
+		time.Sleep(100 * time.Millisecond)
```

Comment:
I think the same strategy could be applied here by testing the function directly without `WaitOn`.